### PR TITLE
[Impeller] implement clear color opt for experimental canvas.

### PR DESCRIPTION
**Pull request by jonahwilliams**
Fixes https://github.com/flutter/flutter/issues/150995

Rather than using a separate state bit, we can use whether the current render pass is active as an indicator for whether or not it is safe to apply the clear color optimization.

### CODE REVIEW

File: impeller/renderer/render_target.cc [hunk 1]
```diff
@@ -209,6 +209,10 @@ const std::optional<StencilAttachment>& RenderTarget::GetStencilAttachment()
   return stencil_;
 }
 
+ColorAttachment& RenderTarget::GetColorAttachment(size_t index) {
```
Comment (chinmaygarde) [inline impeller/renderer/render_target.cc:212]: The `index` is being ignored. You're always going to get 0u.

Comment (chinmaygarde) [reply impeller/renderer/render_target.cc:212]: On second thought, this seems like an unnecessary API change. You can `GetColorAttachments` to fetch the attachment, modify it as necessary, and then `SetColorAttachment`.

**Review by chinmaygarde — CHANGES_REQUESTED** [review]

File: impeller/renderer/render_target.cc (hunk 1 above)
Comment (jonahwilliams) [reply impeller/renderer/render_target.cc:212]: Yeah good point, I would need to modify the map to not be const though

Comment (chinmaygarde) [reply impeller/renderer/render_target.cc:212]: SetColorAttachment should do the trick right?

Comment (jonahwilliams) [reply impeller/renderer/render_target.cc:212]: Yeah that will work!

**Review by chinmaygarde — APPROVED** [review]

**Pull request merged**